recipes/cad1/task2/baseline/car_scene_acoustics.py

```diff
@@ -0,0 +1,260 @@
+        noise_signal = self.carnoise.generate_car_noise(
+            noise_parameters=car_noise_params,
+            number_noise_sources=2,
+            commonness_factor=0,
+        )
+        return noise_signal
+
+    def add_car_hrtf(self, signal: np.ndarray, hrir: dict) -> np.ndarray:
+        """Add a head rotation transfer function using binaural room impulse
+            response (BRIR) from eBrird.
+
+        Args:
+            signal (np.ndarray): a numpy array of shape (2, n_samples) containing the
+                stereo audio signal.
+            hrir: a dictionary containing the HRIR (head-related impulse response) filenames.
+
+        Returns:
+            A numpy array of shape (2, n_samples) containing the stereo audio signal with the
+                BRIR added.
+
+        """
+        car_hrtf_path = Path(self.hrtf_dir) / "Car" / "audio"
+
+        # HRTF from left speaker (LS03) and front mic in HA (CH1)
+        hr_ls03_ch1_left = wavfile.read(
+            car_hrtf_path / f"{hrir['LSP03_CH1_Left']}.wav"
+        )[1]
+        hr_ls03_ch1_right = wavfile.read(
+            car_hrtf_path / f"{hrir['LSP03_CH1_Right']}.wav"
+        )[1]
+
+        # HRTF from right speaker (LS04) and front mic in HA (CH1)
+        hr_ls04_ch1_left = wavfile.read(
+            car_hrtf_path / f"{hrir['LSP04_CH1_Left']}.wav"
+        )[1]
+        hr_ls04_ch1_right = wavfile.read(
+            car_hrtf_path / f"{hrir['LSP04_CH1_Right']}.wav"
+        )[1]
+
+        # add the BRIRs to the signal
+        # Left Speaker (LS03)
+        out_left = lfilter(hr_ls03_ch1_left, 1, signal[0, :])
+        our_right = lfilter(hr_ls03_ch1_right, 1, signal[0, :])
+        # Right Speaker (LS04)
+        out_left += lfilter(hr_ls04_ch1_left, 1, signal[1, :])
+        our_right += lfilter(hr_ls04_ch1_right, 1, signal[1, :])
+
+        return np.stack([out_left, our_right], axis=0)
+
+    def scale_signal_to_snr(
+        self,
+        signal: np.ndarray,
+        reference_signal: np.ndarray = None,
+        snr: Optional[float] = None,
+    ) -> np.ndarray:
+        """
+        Scales the target signal to the desired SNR.
+        We transpose channel because pylodnorm operates
+        on arrays with shape [n_samples, n_channels].
+
+        Args:
+            target_signal (np.ndarray): The target signal to scale.
+            reference_signal (np.ndarray): The reference signal.
+                If None, the reference signal is set to -12 LUFS.
+            snr (float or None): The desired SNR in dB.
+                If None, the target signal is scaled to the reference signal.
+
+        Returns:
+            np.ndarray: The scaled target signal.
+        """
+        # Ensure channels are in the correct dimension
+        if reference_signal.shape[0] < reference_signal.shape[1]:
+            reference_signal = reference_signal.T
+        if signal.shape[0] < signal.shape[1]:
+            signal = signal.T
+
+        if reference_signal is None:
```

Comment:
Could use the same one-liner you have in other places here...

```suggestion
        ref_signal_lufs = -12.0 if reference_signal is None else self.loudness_meter.integrated_loudness(reference_signal)
```